### PR TITLE
feat: Tracking file IO writing

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Added performance tracking for NSData file writing.

Data writing standalone transaction

image

Data writing span inside an ongoing transaction 

image

## :bulb: Motivation and Context

Automatically measure file writing.

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

**Comment by bruno-garcia** [comment]
Would be nice to get a screenshot of what this looks like in Sentry ( a transaction)

**Comment by dashed** [comment]
@brustolin  @rhcarvalho @bruno-garcia would `ok` be a good default for the `status`? `unknown` is considered an error; and could be used whenever the write operation has errored.

File: Sources/Sentry/SentryNSDataTracker.m [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+- (SentrySpanId *)startTrackingForNSData:(NSData *)data filePath:(NSString *)path
+{
+
+    if (![self shouldTrackPath:path])
+        return nil;
+
+    SentrySpanId *spanId = [self.tracker startSpanWithName:[path lastPathComponent]
+                                                 operation:SENTRY_IO_WRITE_OPERATION];
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Sources/Sentry/SentryNSDataTracker.m:54]: In a call, we agreed on 
* Send filename + bytes count as description
* Send full file path as data

**Comment by philipphofmann** [comment]
As we aren't working on this right now @brustolin, let's close it and open up a new PR once we get back to this.

**Pull request closed without merging**